widget/entry.go

```diff
@@ -941,10 +941,27 @@ func (e *Entry) registerShortcut() {
 		e.selecting = false
 		moveWord(se)
 	}
+	removeWord := func(se fyne.Shortcut) {
+		row := e.textProvider().row(e.CursorRow)
+		start, end := getTextWhitespaceRegion(row, e.CursorColumn, true)
+		if start == -1 || end == -1 {
+			return
+		}
+
+		e.propertyLock.Lock()
+		pos := e.cursorTextPos()
+		e.textProvider().deleteFromTo(start, pos)
+		e.CursorRow, e.CursorColumn = e.rowColFromTextPos(pos - 1)
+		e.propertyLock.Unlock()
+
+		e.updateTextAndRefresh(e.textProvider().String())
+	}
+
 	e.shortcut.AddShortcut(&desktop.CustomShortcut{KeyName: fyne.KeyLeft, Modifier: fyne.KeyModifierShortcutDefault}, unselectMoveWord)
 	e.shortcut.AddShortcut(&desktop.CustomShortcut{KeyName: fyne.KeyLeft, Modifier: fyne.KeyModifierShortcutDefault | fyne.KeyModifierShift}, selectMoveWord)
 	e.shortcut.AddShortcut(&desktop.CustomShortcut{KeyName: fyne.KeyRight, Modifier: fyne.KeyModifierShortcutDefault}, unselectMoveWord)
 	e.shortcut.AddShortcut(&desktop.CustomShortcut{KeyName: fyne.KeyRight, Modifier: fyne.KeyModifierShortcutDefault | fyne.KeyModifierShift}, selectMoveWord)
+	e.shortcut.AddShortcut(&desktop.CustomShortcut{KeyName: fyne.KeyBackspace, Modifier: fyne.KeyModifierShortcutDefault}, removeWord)
```

Comment:
On Mac the shortcut for this is Option (alt) + backspace. I think we should try to follow this convention, either using different files with build guards to define the shortcuts, or runtime.GOOS checks

Reply:
Hmm, interesting. I updated it to use the alt shortcut on macOS instead.

Reply:
Interesting find: It seems like Go is smart enough to optimize away runtime.GOOS checks. It is a trivial optimization but nevertheless, I am impressed :)
https://go.godbolt.org/z/b8GYaxz6Y